apps/arena/lib/arena/game_updater.ex

```diff
@@ -342,6 +355,12 @@ defmodule Arena.GameUpdater do
 
     broadcast_player_dead(state.game_state.game_id, victim_id)
 
+    GameTracker.push_event(
+      self(),
+      {:kill, %{id: killer.id, character_name: killer.aditional_info.character_name},
+       %{id: victim.id, character_name: victim.aditional_info.character_name}}
+    )
+
```

Comment:
Why the change of storing the whole match at the end to send every kill event in time?

Reply:
Essentially we keep doing that, but rather than `GameUpdater` keeping track of it we have this specialised module for it. So now we push this events towards it instead of saving it in the game state, a benefit of this pushing is that we can order things as they happen

The main reason for the change was that I had made for when this stats updates where meant to be sent every X seconds during a game, not only at the end. Afterwards I still liked the design cause it allows to decouple things a bit and in the case when `gateway` is down/unabe to process the request we will have the retry logic here. Yes, we could have it in `GameUpdater` instead, but that process is "heavy" as it contains a lot of state so keeping it around only for retries could be problematic for the resource consumptions of the server

Reply:
Perfectly explained. Thank you 😄